nl2sql/expand_on_results.py

```diff
@@ -0,0 +1,43 @@
+from pathlib import Path
+from typing import Union
+from textwrap import dedent
+
+import langchain
+from langchain.cache import SQLiteCache
+from langchain.chat_models import ChatOpenAI
+from langchain.prompts import (ChatPromptTemplate, HumanMessagePromptTemplate,
+                               SystemMessagePromptTemplate)
+
+
+def expand_on_results(docs: Union[str, list], user_query: str, sql_results: list[dict]) -> str:
```

Comment:
I think it would be cleaner if `sql_result` type was a `pd.DataFrame`. It is more common than `list[dict]` especially when passing tabular data.